File: Modul-6/Soal-4/PRAK604-2410817320001-NazlaSalsabila.c
#include <stdio.h>
#include <string.h>

int main() {
    char kode[100], pesan[100], hasil[100];
    int jumlah_bintang = 0, jumlah_pagar = 0;

    printf("Masukkan kode Shikamaru: ");
    scanf(" %[^\n]", kode);
    printf("Masukkan pesan yang diterima: ");
    scanf(" %[^\n]", pesan);

    if (strlen(kode) != strlen(pesan)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
    } else {
        int panjang = strlen(kode);
        for (int i = 0; i < panjang; i++) {
            hasil[i] = (kode[i] == ' ' && pesan[i] == ' ') ? ' ' : 
                       (kode[i] == pesan[i] ? (jumlah_bintang++, '*') : (jumlah_pagar++, '#'));
        }
        hasil[strlen(kode)] = '\0';
        printf("%s\n* = %d\n# = %d\n", hasil, jumlah_bintang, jumlah_pagar);
        printf("%s\n", jumlah_bintang >= jumlah_pagar ? "Pesan Asli" : "Pesan Palsu");
    }

    return 0;
}

File: Modul-6/Soal-4/PRAK604-2410817320001-NazlaSalsabila.py
kode = input("Masukkan kode Shikamaru: ")
pesan = input("Masukkan pesan yang diterima: ")

if len(kode) != len(pesan):
    print("Panjang kalimat berbeda, pesan palsu")
else:
    hasil = []
    jumlah_bintang = 0
    jumlah_pagar = 0

    for k, p in zip(kode, pesan):
        if k == " " and p == " ":
            hasil.append(" ")
        elif k == p:
            hasil.append("*")
            jumlah_bintang += 1
        else:
            hasil.append("#")
            jumlah_pagar += 1

    print("".join(hasil))
    print(f"* = {jumlah_bintang}")
    print(f"# = {jumlah_pagar}")

    if jumlah_bintang >= jumlah_pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")